x/valset-pref/types/expected_interfaces.go

```diff
@@ -17,16 +18,27 @@ type StakingInterface interface {
 	BeginRedelegation(ctx sdk.Context, delAddr sdk.AccAddress, valSrcAddr, valDstAddr sdk.ValAddress, sharesAmount sdk.Dec) (completionTime time.Time, err error)
 	GetDelegatorDelegations(ctx sdk.Context, delegator sdk.AccAddress, maxRetrieve uint16) (delegations []stakingtypes.Delegation)
 	GetValidators(ctx sdk.Context, maxRetrieve uint32) (validators []stakingtypes.Validator)
+	GetParams(ctx sdk.Context) stakingtypes.Params
 }
 
 type BankKeeper interface {
 	GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin
 }
 
-// For testing only
 type DistributionKeeper interface {
 	WithdrawDelegationRewards(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) (sdk.Coins, error)
 	IncrementValidatorPeriod(ctx sdk.Context, val stakingtypes.ValidatorI) uint64
 	CalculateDelegationRewards(ctx sdk.Context, val stakingtypes.ValidatorI, del stakingtypes.DelegationI, endingPeriod uint64) (rewards sdk.DecCoins)
 	AllocateTokensToValidator(ctx sdk.Context, val stakingtypes.ValidatorI, tokens sdk.DecCoins)
 }
+type LockupKeeper interface {
+	GetLockByID(ctx sdk.Context, lockID uint64) (*lockuptypes.PeriodLock, error)
+	HasAnySyntheticLockups(ctx sdk.Context, lockID uint64) bool
+	GetAllSyntheticLockupsByLockup(ctx sdk.Context, lockID uint64) []lockuptypes.SyntheticLock
+	ForceUnlock(ctx sdk.Context, lock lockuptypes.PeriodLock) error
+	BeginUnlock(ctx sdk.Context, lockID uint64, coins sdk.Coins) error
+}
+
+type GammKeeper interface {
+	GetPoolDenoms(ctx sdk.Context, poolId uint64) (denoms []string, err error)
+}
```

Comment:
unused import, I thin Keeper?k we can remove gammKeeper from valset Keeper?

Reply:
yup!